src/inmanta/env.py

```diff
@@ -332,11 +332,19 @@ def _get_requirements_on_inmanta_package(self) -> Sequence[Requirement]:
         to make sure that no Inmanta packages gets overridden.
         """
         workingset: Dict[str, version.Version] = PythonWorkingSet.get_packages_in_working_set()
-        requirements: Sequence[Requirement] = []
-        for pkg in workingset:
-            if pkg == "inmanta" or (pkg.startswith("inmanta-") and not pkg.startswith("inmanta-module-")):
-                requirements.append(Requirement.parse(f"{pkg}=={workingset[pkg]}"))
-        return requirements
+
+        def _is_protected_package(pkg: str) -> bool:
+            """
+            Return true iff the package with name `pkg`, installed in this venv, should not be updated.
+            """
+            if pkg == "inmanta":
+                return True
+            if pkg.startswith("inmanta-") and not pkg.startswith("inmanta-module-") and pkg != "inmanta-dev-dependencies":
```

Comment:
can't we just look at all loaded extensions and protect those?

Reply:
Good idea. I didn't think of that.